tinygrad/codegen/kernel.py

```diff
@@ -389,8 +389,21 @@ def apply_opt(self, opt:Opt, append_opt:bool=True):
                                       (self.group_for_reduces and opt.op not in {OptOps.NOLOCALS, OptOps.PADTO})):
       acc_sz = self.reduceop.dtype.itemsize
       upcast_sz = prod([a for a,b in zip(self.full_shape[self.first_upcast:], self.sts[0].shape[self.first_upcast:]) if a == b])
-      local_sz = prod(self.full_shape[self.first_reduce-self.local_dims:self.first_reduce+self.group_for_reduces])
```

Comment:
when there are multiple grouped reduceops, a local buffer of the same size if made for each to the size of the reduce
(local buffers are made for reduceops in [kernel.py](https://github.com/tinygrad/tinygrad/blob/6a73ad89a2e51b5a14cfe480a2f6d9d9f3f81ad5/tinygrad/codegen/kernel.py#L691) )

so this calculation is no longer correct. We do still need to get the product of `self.full_shape[self.first_reduce-self.local_dims:self.first_reduce`. Buf for `self.full_shape[self.first_reduce:self.first_reduce+self.group_for_reduces]` we need to sum up the products of each reduce's axis

ex. for  `self.full_shape[self.first_reduce:self.first_reduce+self.group_for_reduces] = (4, 4)`  we might have a kernel:
```
extern "C" __global__ void __launch_bounds__(16) r_8_8_2_2_2_2_16_16_8_8(float* data0, const float* data1) {
  __shared__ float temp0[4];
  __shared__ float temp1[4];
...
```
in which `(4 + 4) * sizeof(float)` local memory is used instead of `4 * 4 * sizeof(float)`

Reply:
tensor core optimizations are mutually exclusive with grouping (re [kernel.py](https://github.com/tinygrad/tinygrad/blob/6a73ad89a2e51b5a14cfe480a2f6d9d9f3f81ad5/tinygrad/codegen/kernel.py#L404)) and tensor core optimization also do not affect `self.group_for_reduces`, so this PR shouldn't change anything about how tensor cores are scheduled